py3-pathspec.yaml

```diff
@@ -1,37 +1,53 @@
 package:
   name: py3-pathspec
   version: 0.12.1
-  epoch: 0
+  epoch: 1
   description: "Utility library for gitignore style pattern matching of file paths"
   copyright:
     - license: MPL-2.0
   dependencies:
-    runtime:
-      - python3
+    provider-priority: 0
+
+vars:
+  pypi-package: pathspec
+
+data:
+  - name: py-versions
+    items:
+      3.10: "310"
+      3.11: "311"
+      3.12: "312"
 
 environment:
   contents:
     packages:
       - build-base
       - busybox
-      - ca-certificates-bundle
-      - py3-setuptools
-      - python3
-      - wolfi-base
+      - py3-supported-pip
+      - py3-supported-python
 
 pipeline:
   - uses: fetch
     with:
       uri: https://files.pythonhosted.org/packages/source/p/pathspec/pathspec-${{package.version}}.tar.gz
       expected-sha512: 19d1a8ae5936e76eacd691b0e2fb33d0de376556751a9ae007f6b09d0aa36af7d171101df620cab3e93c9913be578ff3d6bba9d4460351ac248e5c9e015380ad
 
-  - runs: |
-      python3 setup.py build
-
-  - runs: |
-      python3 setup.py install --prefix=/usr --root="${{targets.destdir}}" --skip-build
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: ${{vars.pypi-package}} installed for python${{range.key}}
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+    dependencies:
+      provides:
+        - py3-${{vars.pypi-package}}
+      provider-priority: ${{range.value}}
 
-  - uses: strip
+  - name: py3-wheels-${{vars.pypi-package}}
```

Comment:
We also have a melange test file, there's just a simple `import` to make sure that all the things are there:
https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/python/import.yaml

Just poking around however, I see so many of existing ones that don't have them, so I reckon that can be a project of it's own. WDYT?

Reply:
But, I also think that pipeline will need to be modified to support which version of python it will support, so let's do that first. After this gets merged, I'll add the tests.

Reply:
forking into https://github.com/chainguard-dev/melange/issues/1306